New package: Ubiquiti.UniFiNetworkServer version 8.2.93

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] This PR resolves #158707 

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/158729)

Dragon1573 committed:
New package: Ubiquiti.UniFiNetworkServer version 8.2.93

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
Validated in `amd64` _Windows Sandbox_ instance.

```text
--> Installing WinGet
--> Disabling safety warning when running installer
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.


--> Installing the Manifest 8.2.93

Found Ubiquiti UniFi Network Server [Ubiquiti.UniFiNetworkServer] Version 8.2.93
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://dl.ui.com/unifi/8.2.93/UniFi-installer.exe
  ██████████████████████████████   208 MB /  208 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName                  DisplayVersion Publisher ProductCode    Scope
-----------                  -------------- --------- -----------    -----
Ubiquiti UniFi (remove only)                          Ubiquiti UniFi Machine
```

Dragon1573 commented:
Seems this package requires several TCP/UDP ports opened before first launch. I can't open **Windows Firewall** control panel in _Windows Sandbox_ instance. This may block the initialization of the package.

![image](https://github.com/microsoft/winget-pkgs/assets/49941141/7d362e1d-1934-4ef1-8b78-dba28c1b2d71)

stephengillie commented:
Automatic Validation ended with:


(Automated response - build 886.)

stephengillie commented on manifests/u/Ubiquiti/UniFiNetworkServer/8.2.93/Ubiquiti.UniFiNetworkServer.installer.yaml:
```diff
@@ -0,0 +1,21 @@
+ReleaseDate: 2024-06-03
+AppsAndFeaturesEntries:
+  - DisplayName: Ubiquiti UniFi
+Installers:
```
```suggestion
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x86
Installers:
```

Dragon1573 replied:
Seems it's the actual reason. Verified in my `amd64` _Windows Sandbox_.

![image](https://github.com/microsoft/winget-pkgs/assets/49941141/91ad051a-2f7d-4ceb-8cfc-8ef8278cbe60)
![image](https://github.com/microsoft/winget-pkgs/assets/49941141/560a3a1d-0194-48e2-a8ce-99a736f63846)

stephengillie reviewed (changes requested):
Verify: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/c3ed5e5b-95ac-47cc-af67-d0b85e99c890)
The button didn't become enabled while I was reviewing it, possibly due to:

- This appears to be the [Windows build for MongoDB](https://www.mongodb.com/docs/manual/reference/program/mongod.exe/), and needs `VCRedist`:

![image](https://github.com/microsoft/winget-pkgs/assets/114952565/7399c647-8996-4455-8a1d-2b30fc4de92e)

Application doesn't add version data to Registry:
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/c1f59ce5-1577-4438-bf73-66b9e10e860f)

Dragon1573 committed:
Apply suggestions from code review

- Add dependencies of the package
- This package has a MongoDB bundled in it, but without MSVC Runtime
- We have to add MSVC Runtime to dependencies manually

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
> Template: msftbot/validationError/urls/domain

Can't get it. 😕

stephengillie commented:
- https://github.com/microsoft/winget-pkgs/issues/142854
